### ISSUE TITLE
Implement Autocomplete Feature for Command Entry

### ISSUE THREAD

**Issue by carl0smat3us** [taskipy/taskipy#75]
As a new user of this cli, I found it extremely useful for saving time when typing commands in the terminal. However, I believe we can further improve the user experience by implementing autocomplete functionality for command entry.

Autocomplete would speed up the process of entering commands by suggesting conclusions based on user input, reducing the need for manual typing and minimizing errors. This feature would not only save time but also improve the overall usability of the cli.

**Comment by carl0smat3us** [comment]
And with autocomplete functionality, we can suggest an option when you've already entered an incorrect command.

For example, if you typed:
```sh
task linr
```
We could provide a suggestion like this:
```sh
Could not find task "linr", Did you mean "lint"?
```

**Comment by eugenetriguba** [comment]
@carl0smat3us Hey Carlos, sounds like a great idea!

This reads as two different feature requests so I just wanted to reiterate and confirm this is what you're saying:

- In terms of autocomplete, I'm a little unsure of what exactly you're getting at. Are you referring to shell-based tab completion? I'm most familiar with "autocomplete" behavior in CLIs as being able to press "tab" and it showing a list of options based on the text currently inputted. This is shell dependent and would need some work for each shell desired to be supported.

> And with autocomplete functionality, we can suggest an option when you've already entered an incorrect command.
- So when a user types in a task name that doesn't exist, instead of just "Could not find task abc", you're proposing it should also give a suggestion of the "closest" match (however that would be defined)?

**Comment by carl0smat3us** [comment]
Exactly! In terms of autocomplete, I'm referring to shell-based autocomplete. We can prioritize some shells first

**Comment by carl0smat3us** [comment]
I would like to work on these features.

**Comment by eugenetriguba** [comment]
@carl0smat3us Go for it, thank you 👍

**Comment by grafail** [comment]
This worked pretty nicely for Zsh
```zsh
function _taskipy_tasks {
    local -a tasks
    tasks=($(task -l 2>/dev/null | tail -n +2 | awk '{print $1}'))
    compadd $tasks
}

compdef _taskipy_tasks task
```

**Comment by HenriqueAJNBranco** [comment]
Maybe it's possible to use a tool like [inquirerpy](https://inquirerpy.readthedocs.io/en/latest/) to tackle this =)
